### PR TITLE
CI make sure we run Unit Tests step

### LINKED ISSUE #669: Make sure iOS unit tests are run during CI build
Currently the iOS unit tests are not run during CI build.

We need to run them to avoid regression issues

### PR DESCRIPTION
**Pull request by BPerlakiH**
fixes #669

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -50,3 +50,7 @@ jobs:
           APPLE_DEVELOPMENT_SIGNING_CERTIFICATE: ${{ secrets.APPLE_DEVELOPMENT_SIGNING_CERTIFICATE }}
           APPLE_DEVELOPMENT_SIGNING_P12_PASSWORD: ${{ secrets.APPLE_DEVELOPMENT_SIGNING_P12_PASSWORD }}
           EXTRA_XCODEBUILD: ${{ env.EXTRA_XCODEBUILD }}
+      
+      - name: Run Unit Tests
+        if: matrix.platform == 'macOS'
```
Comment (BPerlakiH) [inline .github/workflows/ci.yml:55]: We do not need to run them twice (iOS + macOS),
they are going to be the same tests as of now.
The macOS will run significantly faster, as it won't require to set up the simulator.

**Comment by BPerlakiH** [comment]
Test run OK, see here: 
https://github.com/kiwix/apple/actions/runs/8031581046/job/21940048205

```
2024-02-24 15:47:42.759 xcodebuild[11285:54834] [MT] IDETestOperationsObserverDebug: 5.961 sec, +5.961 sec -- end
Test session results, code coverage, and logs:
	/Users/runner/Library/Developer/Xcode/DerivedData/Kiwix-egxgvqchbeuqdtfogfzzmwyifluf/Logs/Test/Test-Kiwix-2024.02.24_15-46-37-+0000.xcresult
** TEST SUCCEEDED **
Testing started
```

**Comment by BPerlakiH** [comment]
Tested that it can actually [fail as well](https://github.com/kiwix/apple/actions/runs/8031628222/job/21940146518), as of:
```
Failing tests:
	LibraryRefreshViewModelTest.testNewZimFileAndProperties()
Test session results, code coverage, and logs:
	OPDSParserTests.test()
** TEST FAILED **
	/Users/runner/Library/Developer/Xcode/DerivedData/Kiwix-egxgvqchbeuqdtfogfzzmwyifluf/Logs/Test/Test-Kiwix-2024.02.24_15-58-12-+0000.xcresult
Testing started
Error: Process completed with exit code 65.
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (rgaudin) [reply .github/workflows/ci.yml:55]: I don't think that's a good idea. The point of CI is to trap unexpected regressions. With this we might miss an iOS only regression. And we'll never think of changing this when growing tests.
I don't think slightly slower CI runs are an issue

Comment (BPerlakiH) [reply .github/workflows/ci.yml]: Ok, I've updated it to run it on both platforms.

**Review by rgaudin — DISMISSED** [review]
👍

**Comment by BPerlakiH** [comment]
Slight modifications added to make sure it's actually picks up the iOS simulator. It turned out without a 'name' it's not capable to launch the simulator...

**Review by rgaudin — APPROVED** [review]
LGTM

**Pull request merged**